near-sdk/src/types/event.rs

```diff
@@ -0,0 +1,84 @@
+pub trait StandardEvent {
+    type EventString: std::fmt::Display;
+    fn format(&self) -> Self::EventString;
+    fn emit(&self) {
+        crate::env::log_str(&format!("EVENT_JSON:{}", self.format()));
+    }
```

Comment:
Sorry, I initially mentioned `std::fmt::Display` as an alternative to `AsRef<str>` but thinking about it more, it doesn't make much sense. This is because it forces `fmt` machinery in, even if someone wanted to have a static event be emitted.

```suggestion
    type EventString: std::convert::AsRef<str>;
    fn format(&self) -> Self::EventString;
    fn emit(&self) {
        format!("EVENT_JSON:{}", self.format().as_ref());
    }
```